frontend/app/blueprints/groups/templates/group.html

```diff
@@ -130,90 +119,85 @@ <h4 class="card-title me-auto fw-light">{{group_name}}</h4>
+    async function openTreeView ({newick, typingData}) {
         // Open a tree view as a new tab my creating a form and submit it
         const form = document.getElementById("open-tree-form")
         // add metadata and newick file to form
         const newickInput = form.querySelector("#newick-content")
         newickInput.setAttribute("value", newick)
         const metadataInput = form.querySelector("#metadata-content")
+        const typingDataInput = form.querySelector("#typing-data-content")
+        typingDataInput.setAttribute("value", typingData)
         metadataInput.setAttribute("value", 
             window.sessionStorage.getItem("clusteredSamples")
         )
         const submitBnt = form.querySelector("input[type=submit]")
         submitBnt.click()
     }
+
     // add submit clustering function
     document
     .getElementById("cluster-button")
     .addEventListener("click", (event) => {
         event.preventDefault()  // prevent normal behaviour
         // show spinner
         const button = document.getElementById("cluster-button")
-        button.querySelector("#cluster-btn-text").innerText = "Clustering..."
-        const spinner = button.querySelector("#cluster-btn-spinner")
-        spinner.hidden = false
+        showSpinner(button)
         
         // get sample ids for checked samples
         const sampleId = window.getSelectedRows()
-        const typingData = document.getElementById("cluster-typing-data").value
         const distance = document.getElementById("cluster-distance-metric").value
-        const method = document.getElementById("cluster-clustering-method").value
+        // if there are more than 1k samples use minhash instead of cgmlst
+        const typingData = sampleId.length > 1000 ? "minhash" : document.getElementById("cluster-typing-data").value
+        // minhash doesnt support neigbour joining
+        const method = typingData === "minhash" ? "single" : document.getElementById("cluster-clustering-method").value
         // setup and submit clustering job
         const xhr = new XMLHttpRequest()
-        const apiURL = "{{ config.BONSAI_API_URL }}"
         const entrypointURL = `${apiURL}/cluster/${typingData}`
         // add event listeners for processing different outcomes of request
         xhr.addEventListener("load", event => {
             // successful request
             if(xhr.readyState == 4 && xhr.status == 201) {
-                // todo add default parameters
                 // instead open a new tab by mocking a submission
-                openTreeView({newick: xhr.responseText, metadata: "not so much"})
-                spinner.hidden = true
-                button.querySelector("#cluster-btn-text").innerText = orgButtonText
+                const jobId = JSON.parse(xhr.responseText)
+                try {
+                    (async () => {
+                        // start polling for cluster results
+                        let newick = await poll(
```

Comment:
I have seen a similar structure in multiple places within this PR. Would it make sense to extract a helper function from all/parts of this code? (Just wondering)